File: C++/Medium/Accepted/largest-number_1.cpp
class Solution {
public:
    string largestNumber(vector<int>& nums) {
        
        sort(nums.begin(), nums.end(), [](int a, int b) {
            string sa = to_string(a);
            string sb = to_string(b);
            return sa + sb > sb + sa;
        });
        string ans;
        for (int i = 0; i < nums.size(); i++) {
            ans += to_string(nums[i]);
        }
        if (ans[0] == '0') {
            return "0";
        }
        return ans;
    }
};

File: Rust/Medium/Accepted/largest-number_1.rs
impl Solution {
    pub fn largest_number(nums: Vec<i32>) -> String {
        

        let mut nums = nums;
        nums.sort_by(|a, b| {
            let a = a.to_string();
            let b = b.to_string();
            let ab = a.clone() + &b;
            let ba = b + &a;
            ba.cmp(&ab)
        });
        let mut ans = nums
            .into_iter()
            .map(|x| x.to_string())
            .collect::<Vec<String>>()
            .join("");
        if ans.chars().next().unwrap() == '0' {
            ans = "0".to_string()
        }
        ans
    }
}